corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js

```diff
@@ -162,13 +162,19 @@ Container.prototype.fromJS = function(json) {
                 if (ko.utils.unwrapObservable(options.target.serverError)) {
                     options.data.answer = _.clone(options.target.answer());
                 }
+                if (_.isEqual(options.target.choices(), options.data.choices)) {
```

Comment:
Is there anything like an id or path in the array elements that you could pluck out to make the comparison quicker?

Reply:
Ha, so actually choices is an array of strings, not of the richer objects you might expect, so there's nothing to compare on but these string values themselves.